tests/integration/conftest.py

```diff
@@ -1,32 +1,151 @@
     metadata = Path("./metadata.yaml")
     data = yaml.safe_load(metadata.read_text())
     return data
 
 
-@pytest.fixture
-def model(ops_test):
+@pytest.fixture(scope="module")
+def app_name() -> str:
+    # Randomized app name to avoid collision when connecting to GitHub.
+    return f"integration-{secrets.token_hex(4)}"
+
+
+@pytest.fixture(scope="module")
+def path(pytestconfig: pytest.Config) -> str:
+    path = pytestconfig.getoption("--path")
+    assert path, "Please specify the --path command line option"
+    return path
+
+
+@pytest.fixture(scope="module")
+def token(pytestconfig: pytest.Config) -> str:
+    token = pytestconfig.getoption("--token")
+    assert token, "Please specify the --token command line option"
+    return token
+
+
+@pytest.fixture(scope="module")
+def token_alt(pytestconfig: pytest.Config, token: str) -> str:
+    token_alt = pytestconfig.getoption("--token-alt")
+    assert token_alt, "Please specify the --token-alt command line option"
+    assert token_alt != token, "Please specify a different token for --token-alt"
+    return token_alt
+
+
+@pytest.fixture(scope="module")
+def http_proxy(pytestconfig: pytest.Config) -> str:
+    http_proxy = pytestconfig.getoption("--http-proxy")
+    return "" if http_proxy is None else http_proxy
+
+
+@pytest.fixture(scope="module")
+def https_proxy(pytestconfig: pytest.Config) -> str:
+    https_proxy = pytestconfig.getoption("--https-proxy")
+    return "" if https_proxy is None else https_proxy
+
+
+@pytest.fixture(scope="module")
+def no_proxy(pytestconfig: pytest.Config) -> str:
+    no_proxy = pytestconfig.getoption("--no-proxy")
+    return "" if no_proxy is None else no_proxy
+
+
+@pytest.fixture(scope="module")
+def model(ops_test: OpsTest) -> Model:
+    assert ops_test.model is not None
     return ops_test.model
 
 
-@pytest.fixture
-def application(model, metadata):
-    charm_name = metadata["name"]
-    app = model.applications[charm_name]
-    return app
+@pytest_asyncio.fixture(scope="module")
+async def lxd_profile() -> AsyncIterator[Path]:
+    lxd_profile_path = Path("lxd-profile.yaml")
+
+    with open(lxd_profile_path, "w") as profile_file:
+        profile_file.writelines(
+            """config:
+    security.nesting: true
+    security.privileged: true
+    raw.lxc: |
+        lxc.apparmor.profile=unconfined
+        lxc.mount.auto=proc:rw sys:rw cgroup:rw
+        lxc.cgroup.devices.allow=a
+        lxc.cap.drop=
+devices:
+    kmsg:
+        path: /dev/kmsg
+        source: /dev/kmsg
+        type: unix-char
+"""
+        )
+
+    yield lxd_profile_path
+
+    lxd_profile_path.unlink(missing_ok=True)
+
+
+@pytest_asyncio.fixture(scope="module")
+async def app_no_token(
+    ops_test: OpsTest,
+    model: Model,
+    app_name: str,
+    path: str,
+    http_proxy: str,
+    https_proxy: str,
+    no_proxy: str,
+    lxd_profile: Path,
+) -> AsyncIterator[Application]:
+    charm = await ops_test.build_charm(".")
```

Comment:
I think the new operator workflows injects the built charm via [`--charm-file` argument](https://github.com/canonical/operator-workflows/blob/33a71e5b6fccdc9af17cd4cd4c1a92fdf3171e51/.github/workflows/integration_test_run.yaml#L226) which can be used to reduce a bit of build time :D

Reply:
`operator-workflow` is not used as GitHub secrets needs to be passed to the charm.